test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -88,3 +94,186 @@ def test_invalid_script_out(self):
         self.scheduler.log_match(
             "Insufficient amount of server resource: %s (R: 1 A: 0 T: 0)"
             % (resname))
+
+        self.du.rm(path=tmpfilepath, sudo=True)
+
+    def test_res_long_pos(self):
+        """
+        Test that server_dyn_res accepts command line arguments to the
+        commands it runs. Resource value set to a positive long int.
+        """
+        # Create a resource of type long. positive value
+        resname = ["foobar"]
+        restype = "long"
+        resval = ['"foobar !/bin/echo 4"']
+
+        # Add server_dyn_res entry in sched_config
+        self.setup_dyn_res(resname, restype, resval)
+
+        a = {'Resource_List.foobar': 4}
+        jid = self.submit_job(a)
+
+        # Job must run successfully
+        a = {'job_state': 'R', 'Resource_List.foobar': '4'}
+        self.server.expect(JOB, a, id=jid)
+
+    def test_res_long_neg(self):
+        """
+        Test that server_dyn_res accepts command line arguments to the
+        commands it runs. Resource value set to a negative long int.
+        """
+        # Create a resource of type long. negative value
+        resname = ["foobar"]
+        restype = "long"
+        resval = ['"foobar !/bin/echo -1"']
+
+        # Add server_dyn_res entry in sched_config
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Submit job
+        a = {'Resource_List.foobar': '1'}
+        jid = self.submit_job(a)
+
+        # Check for the expected log message for insufficient resources
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foobar (R: 1 A: -1 T: -1)"
+
+        # The job shouldn't run
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_whitespace(self):
+        """
+        Test for parse errors when more than one white space
+        is added between the resource name and the !<script> in a
+        server_dyn_res line. There shouldn't be any errors.
+        """
+        # Create a resource of type long
+        resname = ["foo"]
+        restype = "long"
+
+        # Prep for server_dyn_resource scripts. Script "PbsPtl_get_foo*"
+        # generates file "PbsPtl_got_foo" and returns 1.
+        script_body = "echo get_foo > /tmp/PbsPtl_got_foo; echo 1"
+
+        fpath_out = os.path.join(os.sep, "tmp", "PbsPtl_got_foo")
+
+        (fd_in, fn_in) = self.du.mkstemp(prefix="PtlPbs_get_foo",
+                                         body=script_body)
+        self.du.chmod(path=fn_in, mode=0755, sudo=True)
+        os.close(fd_in)
+
+        # Add additional white space between resource name and the script
+        resval = ['"' + resname[0] + ' !' + fn_in + '"']
+
+        self.setup_dyn_res(resname, restype, resval)
+
+        # Check if the file "PbsPtl_got_foo" was created
+        time.sleep(15)
```

Comment:
why are you sleeping for this long?

Reply:
The file gets generated when the scheduling cycle starts and executes the server dynamic resource script. On my test machine, I typically have to wait for 3 sec. I had added additional buffer (15s total, may be more!) to account for slower test machines.  I have changed it to 5 sec now. Will revisit later to increase the sleep time it fails on slower machines.